test/bidirectional_graph_test.rb

```diff
@@ -130,17 +121,25 @@ def test_add_vertices
   end
 
   def test_creating_from_array
-    dg = BidirectionalAdjacencyGraph[1, 2, 3, 4]
+    dg = @gfa.clone
     assert_equal([1, 2, 3, 4], dg.vertices.sort)
     assert_equal('(1-2)(3-4)', dg.edges.join)
   end
 
+  def test_creating_from_graphs
+    dg = @dg.clone
+    @gfa.each_edge { |e| dg.add_edge(e[0], e[1])}
+    dg2 = BidirectionalAdjacencyGraph.new(Set, @dg, @gfa)
+    assert_equal(dg2, dg2)
```

Comment:
Do we test `identity` here (see [Graph#eql?](https://monora.github.io/rgl/RGL/Graph.html#eql%3F-instance_method)), because you compare the same variable `dg2`?

A better test would be:

```ruby
assert_equal(dg2.vertices.sort, (@dg.vertices + @gfa.vertices).sort)
assert_equal(dg2.edges.sort.join, (@dg.edges + @gfa.edges).sort.join)
```

Reply:
The test in that form fails because `@dg` and `@dfa` contain duplicated vertices. It works if we replace `.sort` and `.sort.join` with `.to_set`.

Reply:
Order independence is in the specification for `Set#eql?`: https://github.com/ruby/spec/blob/master/library/set/eql_spec.rb

Reply:
Ah, didn't know that. Then we can omit sorting!

Reply:
I would now squash, rebase and release. Thank you for your contribution!